ember-scoped-css/src/lib/scoped-css-preprocessor.js

```diff
@@ -67,12 +72,118 @@ class ScopedFilter extends Filter {
       return '';
     }
   }
+
+  async postProcess(results, relativePath) {
+    if (process.env.CI || relativePath.endsWith('.module.css')) {
+      return results;
+    }
+
+    for (let inputPath of this.inputPaths) {
+      const templateFile = {};
+
+      for (let ext of TEMPLATE_EXTENSIONS) {
+        const templatePath = relativePath.replace(/\.css/, '.' + ext);
+        const templateFilePath = path.join(inputPath, templatePath);
+        const exists = await fsExists(templateFilePath);
+
+        if (exists) {
+          templateFile.path = templateFilePath;
+          templateFile.ext = ext;
+          continue;
+        }
+      }
+
+      // if the template exists we check the css for changes
+      if (templateFile.path) {
+        const cssFilePath = path.join(inputPath, relativePath);
+        const cssContents = await readFile(cssFilePath, 'utf-8');
+        const { classes, tags } = getClassesTagsFromCss(cssContents);
+        const previousClasses = this.options.previousClasses.get(relativePath);
+
+        // if we have previous classes, and they are different, build templates to compare
+        if (previousClasses && classes !== previousClasses) {
+          const localPackagerStylePath =
+            packageScopedPathToModulePath(relativePath);
+          const postfix = generateHash(localPackagerStylePath);
+          const templateRaw = await readFile(templateFile.path, 'utf-8');
+          const templateComparison = [];
+          let templateContents = templateRaw;
+
+          if (templateFile.ext === 'hbs') {
+            templateComparison.push(
+              didTemplateChange(
+                templateContents,
+                previousClasses,
+                classes,
+                tags,
+                postfix
+              )
+            );
+          } else {
+            // find all template tags, and extract the contents to compare
+            const templates = parseTemplates(templateRaw, '');
```

Comment:
parses templates with `ember-template-tag`, which returns an array of identified template tag objects, with `contents` representing the template markup. We can then use this for the template change check and if any did, we append a space to the file to kick the cache.